src/main/public/js/postcode-lookup.js

```diff
@@ -0,0 +1,261 @@
+(function () {
+  var PostcodeLookup = function() {
+    var widget
+    var searchContainer
+    var searchButton
+    var addressDropdownContainer
+    var addressDropdown
+    var addressContainer
+    var enterAddressManuallyLink
+    var stateHolder
+
+    var State = {
+      PRISTINE: 'PRISTINE',
+      ADDRESS_FOUND: 'ADDRESS_FOUND',
+      ADDRESS_ENTERED: 'ADDRESS_ENTERED'
+    }
+
+    var Error = {
+      MISSING_POSTCODE_ERROR: 'MISSING_POSTCODE_ERROR',
+      UNSUPPORTED_POSTCODE_ERROR: 'UNSUPPORTED_POSTCODE_ERROR',
+      MISSING_ADDRESS_ERROR: 'MISSING_ADDRESS_ERROR',
+      UNKNOWN_ERROR: 'UNKNOWN_ERROR'
+    }
+
+    // Private methods
+
+    function show(element) {
+      element.classList.remove('hidden', 'js-hidden')
+    }
+
+    function hide(element) {
+      element.classList.add('hidden')
+    }
+
+    function showError (errorCode, global) {
```

Comment:
The errors flash on the screen quite badly if you enter a name but not a valid postcode and click enter,
please either make it render blocking or revert back to doing global error messages entirely serverside
REQ